### PR TITLE
#2142. Add more static analysis tests

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

File: LanguageFeatures/Extension-types/static_analysis_extension_types_A02_t01.dart [hunk 1]
```diff
@@ -22,19 +22,19 @@ extension type ET1(int id) {
 // [analyzer] unspecified
 // [cfe] unspecified
 
-  ET1.n2(int id) : this();
+  ET1.n2(int id) : this(id);
```
Comment (chloestefantsova) [inline LanguageFeatures/Extension-types/static_analysis_extension_types_A02_t01.dart:25]: This case doesn't look like it's a compile-time error. Constructor `ET1.n2` declares a formal parameter, `id`, and passes it as the argument to the constructor that initializes the field with the same name. By analogy, the following Dart code compiles and runs without errors:

```dart
class A {
  int foo;
  A(this.foo);
  A.named(int foo): this(foo);
}

main() {}
```

/cc @eernstg

File: LanguageFeatures/Extension-types/static_analysis_extension_types_A02_t01.dart [hunk 2]
```diff
@@ -22,19 +22,19 @@ extension type ET1(int id) {
...
 //^^^
 // [analyzer] unspecified
 // [cfe] unspecified
 
-  ET1.n3(int id) : this.new();
+  ET1.n3(int id) : this.new(id);
```
Comment (chloestefantsova) [inline LanguageFeatures/Extension-types/static_analysis_extension_types_A02_t01.dart:30]: Similarly here. This line doesn't look like a compile-time error to me.

**Review by chloestefantsova — CHANGES_REQUESTED** [review]

File: LanguageFeatures/Extension-types/static_analysis_extension_types_A02_t01.dart (hunk 1 above)
Comment (eernstg) [reply LanguageFeatures/Extension-types/static_analysis_extension_types_A02_t01.dart:25]: I also don't see any compile-time errors here.

File: LanguageFeatures/Extension-types/static_analysis_extension_types_A02_t01.dart (hunk 2 above)
Comment (eernstg) [reply LanguageFeatures/Extension-types/static_analysis_extension_types_A02_t01.dart:30]: Agreed again.

**Review by eernstg — COMMENTED** [review]
Agreeing. ;-)

**Comment by eernstg** [comment]
What's the next step with this PR?

**Comment by sgrekhov** [comment]
Thank you!  It's an error itself to declare constructor named `new` in that test. Fixed. Please take another look

**Comment by eernstg** [comment]
I think it's safer to use the terminology which is also used in the language specification: `class C { C(); C.named(); }` declares a constructor named `C` and another constructor named `C.named`. There is no such thing as a nameless or an anonymous constructor, constructors have names of the form `<typeIdentifier> ('.' <identifier>)?`.

If we use that terminology then we can only have a constructor named `new` in a class named `new`, which is a syntax error because `new` is a reserved word. Also, if we're considering a constructor declaration of the form `ET.new(int i)` (which could be a primary constructor or a non-primary constructor) then its name is `ET`, not `ET.new`. So a constructor named `new` is not really a thing.

Also, a primary as well as a non-primary constructor can have this kind of declaration. (For the "primary" constructor it's currently specified as `<typeName> <typeParameters>? <representationDeclaration>`, but the `<representationDeclaration>` does allow `new` as in `ET.new(int i)`.)

So why would we have this error?:

> It's an error itself to declare constructor named `new`

**Comment by eernstg** [comment]
Ah, OK! Thanks!

**Review by chloestefantsova — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**